backend/src/test/java/com/hackner/musicompass/controller/RecommendationControllerTest.java

```diff
@@ -0,0 +1,122 @@
+import static org.mockito.Mockito.when;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class RecommendationControllerTest {
+
+    @LocalServerPort
+    private int port;
+
+    private String getUrl() {return "http://localhost:" + port + "api/recommendation";}
+
+    @Autowired
+    private TestRestTemplate testRestTemplate;
+
+    @Autowired
+    private ArtistMongoDb artistMongoDb;
+
+    @Autowired
+    private RecommendationCategoryMongoDb recommendationCategoryMongoDb;
+
+    @BeforeEach
+    public void setup(){artistMongoDb.deleteAll();}
+
+    @Test
+    @DisplayName("Get recommendation category values")
+    public void getRecommendationCategoryValues(){
+        //GIVEN
+        recommendationCategoryMongoDb.save(RecommendationCategory.builder().categoryName("Genre").categoryValues(Arrays.asList("Genre1", "Genre2")).build());
+        recommendationCategoryMongoDb.save(RecommendationCategory.builder().categoryName("Role").categoryValues(Arrays.asList("Role1","Role2")).build());
+
+        //WHEN
+        ResponseEntity <List<RecommendationCategory>> response = testRestTemplate.exchange(getUrl(),
+                HttpMethod.GET,
+                null,
+                new ParameterizedTypeReference<List<RecommendationCategory>>() {});
+
+        //THEN
+        assertThat(response.getBody(), hasItem(RecommendationCategory.builder()
+                .categoryName("Genre")
+                .categoryValues(Arrays.asList("Genre1", "Genre2")).build()));
+    }
+
+    @Test
+    @DisplayName("Change recommendation tags for already existing recommendation tags")
+    public void changeExistingRecommendationTags(){
+        //GIVEN
+        String artistName ="artistName";
+        List<String> genresBefore = Arrays.asList("Genre1", "Genre2");
+        List<String> rolesBefore = Arrays.asList("Role1","Role2");
+        String genderBefore = "Gender";
+        List<String> genresAfter = Arrays.asList("Genre1", "Genre5");
+        List<String> rolesAfter = Arrays.asList("Role0","Role2");
+        String genderAfter = "differentGender";
+        Instant before = Instant.ofEpochSecond(Instant.now().getEpochSecond());
+
+        RecommendationTagsDto recommendationTagsDto = RecommendationTagsDto.builder()
+                .artistName(artistName)
+                .genres(genresAfter)
+                .roles(rolesAfter)
+                .gender(genderAfter).build();
+
+        RecommendationTags recommendationTagsBefore = RecommendationTags.builder()
+                .recommended(true)
+                .genres(genresBefore)
+                .roles(rolesBefore)
+                .changeDate(Date.from(before))
+                .gender(genderBefore).build();
+
+        RecommendationTags recommendationTagsAfter = RecommendationTags.builder()
+                .recommended(true)
+                .genres(genresAfter)
+                .roles(rolesAfter)
+                .gender(genderAfter).build();
```

Comment:
A lot of buildup to call a simple method, maybe extract some creation methods?!

Reply:
I agree, I want to look into this again, but so far I didnt figure out a good way to source this out..